### PR TITLE
[PA-57] Reset password feature

### PR DESCRIPTION
**Pull request by leitaku**
checking user's security question and answer done

### CODE REVIEW

File: study_hour/react-client/src/Components/App.jsx [hunk 1]
```diff
@@ -66,6 +67,9 @@ class App extends Component {
                         <Route path="/Location/:id" component={Location}/>
                         <Route path="/Location" component={Location}/>
                         <Route path="/AddLocation" component={AddLocation}/>
+                        <Route path="/ValidateUser" component={ValidateUser}/>
+                        <Route path="/SecurityCheck" component={SecurityCheck}/>
+                        <Route path="/ResetPassword" component={ResetPassword} />
```
Comment (ElliotVilhelm) [inline study_hour/react-client/src/Components/App.jsx]: can we wrap NoAuth components in NoAuth wrapper?

File: study_hour/react-client/src/Components/SecurityCheck.jsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        )
+    }
+}
+function mapStateToProps(state) {
+    return {
+        authenticated: state
+    }
+}
+
+export default connect(mapStateToProps)(withStyles(styles)(withRouter(SecurityCheck)));
```
Comment (ElliotVilhelm) [inline study_hour/react-client/src/Components/SecurityCheck.jsx]: new line pls

File: study_hour/server/index.js [hunk 3]
```diff
@@ -130,6 +130,31 @@ app.post('/api/ListResult', function (req, res, next) {
...
+
+app.post('/api/savePassword', function (req, res, next) {
+    pgClient.query('UPDATE users SET password = $2 WHERE id = $1', [req.body.id, req.body.password], function (err, result) {
+        if (err) {
+            return next(err)
+        }
+        res.send({dbresponse: result.rows})
+    });
+});
+// app.get('/api/ValidateUser', function (req, res, next) {
```
Comment (ElliotVilhelm) [inline study_hour/server/index.js]: delete this?

File: study_hour/react-client/src/Components/App.jsx (hunk 1 above)
Comment (leitaku) [reply study_hour/react-client/src/Components/App.jsx]: LOOK!

File: study_hour/react-client/src/Components/SecurityCheck.jsx (hunk 2 above)
Comment (leitaku) [reply study_hour/react-client/src/Components/SecurityCheck.jsx]: LOOK PLEASE!

File: study_hour/react-client/src/Components/App.jsx [hunk 4]
```diff
@@ -66,6 +67,9 @@ class App extends Component {
                         <Route path="/Location/:id" component={Location}/>
```
Comment (ElliotVilhelm) [inline study_hour/react-client/src/Components/App.jsx]: these too pls

File: study_hour/react-client/src/Components/App.jsx [hunk 5]
```diff
@@ -66,6 +67,9 @@ class App extends Component {
...
                         <Route path="/Location" component={Location}/>
```
Comment (ElliotVilhelm) [inline study_hour/react-client/src/Components/App.jsx]: delete this?

**Pull request merged**